src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+/**
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.net.SocketException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.commons.configuration.ConversionException;
+import org.apache.commons.httpclient.InvalidRedirectLocationException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+/**
+ * The SQLInjectionMsSQL plugin identifies MsSQL specific SQL Injection vulnerabilities
+ * using MsSQL specific syntax.  If it doesn't use MsSQL specific syntax, it belongs in the generic SQLInjection class!
+ * Note the ordering of checks, for efficiency is :
+ * 1) Error based (N/A)
+ * 2) Boolean Based (N/A - uses standard syntax)
+ * 3) UNION based (N/A - uses standard syntax)
+ * 4) Stacked (N/A - uses standard syntax)
+ * 5) Blind/Time Based (Yes - uses specific syntax)
+ *
+ * See the following for some great MySQL specific tricks which could be integrated here
+ * http://www.websec.ca/kb/sql_injection#MSSQL_Stacked_Queries
+ * http://pentestmonkey.net/cheat-sheet/sql-injection/mssql-sql-injection-cheat-sheet
+ */
+public class SQLInjectionMsSQL extends AbstractAppParamPlugin {
+
+	private boolean doTimeBased = false;
+	private int doTimeMaxRequests = 0;
+	private int sleep = 5;
+
+	/**
+	 * MSSQL one-line comment
+	 */
+	public static final String SQL_ONE_LINE_COMMENT = " -- ";
```

Comment:
- Can be private?
 - Instance variables should be moved to after the class variables.